app/src/main/java/com/tomaszpolanski/androidsandbox/MainViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.tomaszpolanski.androidsandbox
+
+import android.annotation.SuppressLint
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Parcelable
+import android.support.v4.app.FragmentActivity
+import io.reactivex.Single
+import kotlinx.android.parcel.Parcelize
+
+
+private const val MAX_CLICK_COUNT = 10
+
+sealed class Event {
+    object LongClick : Event()
+    object ShortClick : Event()
+}
+
+@SuppressLint("ParcelCreator")
+sealed class State : Parcelable {
+    abstract val totalClicks: Int
+
+    @Parcelize data class HiddenToast(override val totalClicks: Int) : State()
+    @Parcelize data class ShownToast(override val totalClicks: Int) : State() {
+        val text get() = "You are ${MAX_CLICK_COUNT - totalClicks} click(s) away to be Groupon dev"
+    }
+}
+
+class MainViewModel(
```

Comment:
Generally, how do you plan to handle operations which last the lifetime of the Activity. What subscription keeps them running when the Activity unsubscribes from the ViewModel?

Reply:
Good point, haven't yet thought about it